nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py

```diff
@@ -52,22 +53,36 @@ def add_recording_extractor_properties(recording_extractor: se.RecordingExtracto
     group_nums = [n + 1 for n, channels in enumerate(channel_groups) for _ in channels]
     group_names = [f"Group{n}" for n in group_nums]
 
-    for channel_id in channel_map.keys():
-        recording_extractor.set_channel_groups(channel_ids=[channel_id], groups=group_nums[channel_map[channel_id]])
-        recording_extractor.set_channel_property(
-            channel_id=channel_id, property_name="group_name", value=group_names[channel_map[channel_id]]
-        )
-        recording_extractor.set_channel_property(
-            channel_id=channel_id,
-            property_name="shank_electrode_number",
-            value=group_electrode_numbers[channel_map[channel_id]],
-        )
+    channel_id_list = list(channel_map.keys())
+    if isinstance(recording_extractor, se.RecordingExtractor):
+        for channel_id in channel_id_list:
+            recording_extractor.set_channel_groups(channel_ids=[channel_id], groups=group_nums[channel_map[channel_id]])
+            recording_extractor.set_channel_property(
+                channel_id=channel_id, property_name="group_name", value=group_names[channel_map[channel_id]]
+            )
+            recording_extractor.set_channel_property(
+                channel_id=channel_id,
+                property_name="shank_electrode_number",
+                value=group_electrode_numbers[channel_map[channel_id]],
+            )
+    else:
+        # Channel groups
+
+        # Group names
+        group_names = [group_names[channel_map[channel_id]] for channel_id in channel_id_list]
+        recording_extractor.set_property(key="group_name", ids=channel_id_list, values=group_names)
+        # Shank electrode number
+        group_electrode_numbers = [group_electrode_numbers[channel_map[channel_id]] for channel_id in channel_id_list]
+        recording_extractor.set_property(key="shank_electrode_number", ids=channel_id_list, values=group_names)
```

Comment:
For blocks like this I'd rather just coerce the object to a `OldtoNew` instead of maintaining the entire logic compatible with the old version. So the logic ought only reflect `BaseRecording` objects, and we obtain compatability with previous version purely through SI and their `old_api_utils`